### PR TITLE
fix(#1236) Tabs updates per accessibility audit

### LINKED ISSUE #1236: Tabs accessibility audit
# 🐛 Bug Report 

Container element with `role="tablist"` is not interactive and should not be focusable.

## 🤔 Expected Behavior

1. Container element with `role="tablist"` should not have a `tabIndex` attribute.
2. Container element with `role="tablist"` should not have the `aria-disabled` attribute.
3. `role="tab"` elements that are not disabled should not include `aria-disabled="false"`.
4. Selected and not disabled `role="tab"` element should have `tabIndex={0}` by default when the tablist initializes.
5. Add aria-label to storybook examples, so that screen readers do not announce the full tablist content as context when focus enters the tablist.
6. Container element with `role="tablist"` should have `aria-orientation="vertical"` when the orientation is vertical.

## 😯 Current Behavior

1. Container element with `role="tablist"`, seems to use `useFocusWithin` to update a `tabIndex` attribute. The CollectionManager then moves focus to the appropriate Tab for the selected key: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/tabs/src/useTabs.ts#L67-L71. The `tablist` role is not interactive, and shouldn't really be focusable, even though it loses focus immediately.
2. Container element with `role="tablist"` includes the `aria-disabled` attribute, even though it is not interactive.
3. `role="tab"` elements that are not disabled include `aria-disabled="false"`.
4. Selected and not disabled `role="tab"` element initializes with `tabIndex={-1}`, that only changes on focus within.
5. In storybook examples, so that screen readers announce the full tablist content as context when focus enters the tablist.
6. Container element with `role="tablist"` does not communicate the orientation to screen reader users when the orientation is vertical, which will present confusion because the keyboard navigation behaviors differ.

## 💁 Possible Solution

PR with suggested updates: #1237

## 🔦 Context

Review Tabs for accessibility.

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |   [v3.0.0-alpha.1 ](https://reactspectrum.blob.core.windows.net/reactspectrum/879cbbdd5031a8d6ac4e91272c8e73d7dd38b0c9/storybook-17/index.html?path=/story/tabs--default)
| Browser          | Chrome Version 86.0.4240.183 (Official Build) (x86_64)
| Operating System | macOS 10.15.7 (19H2)

## 🧢 Your Company/Team

Adobe/Accessibility

### PR DESCRIPTION
**Pull request by majornista**
1. Container element with role="tablist" should not have a tabIndex attribute.
2. Container element with role="tablist" should not have the aria-disabled attribute.
role="tab" elements that are not disabled should not include aria-disabled="false".
3. Selected and not disabled role="tab" element should have tabIndex={0} by default when the tablist initializes.
4. Add aria-label to storybook examples, so that screen readers do not announce the full tablist content as context when focus enters the tablist.

Co-Authored-By: Robert Snow <rsnow@adobe.com>

Closes #1236

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue: #1236.
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Container element with role="tablist" should not have a tabIndex attribute.
2. Container element with role="tablist" should not have the aria-disabled attribute.
role="tab" elements that are not disabled should not include aria-disabled="false".
3. Selected and not disabled role="tab" element should have tabIndex={0} by default when the tablist initializes.
4. Add aria-label to storybook examples, so that screen readers do not announce the full tablist content as context when focus enters the tablist.

## 🧢 Your Project:

Adobe/Accessibility

### CODE REVIEW

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js [hunk 1]
```diff
@@ -22,74 +22,74 @@ import {Tabs} from '..';
 storiesOf('Tabs', module)
   .add(
     'Default',
-    () => render()
+    () => render({'aria-label': 'Default'})
```
Comment (devongovett) [inline packages/@react-spectrum/tabs/stories/Tabs.stories.js:25]: We should probably add a console warning if no label is provided like we do for other components to make this easier to catch.

**Review by snowystinger — COMMENTED** [review]
I'm not sure why, but in 'Default' story, however I select "Tab 1" the onChange says empty string instead of 'val1'
It appears to be a regression because I see the correct behavior on main.

I managed to get into a state where I could tab to the tabs at all. I could get to the content, but not the tabs. I can't make it happen again on purpose yet, I'll let you know if I do manage to reproduce it, but figured I'd bring it up anyways.

File: packages/@react-aria/tabs/src/useTabs.ts [hunk 2]
```diff
@@ -64,19 +63,12 @@ export function useTabs<T>(props: TabsAriaProps<T>, state: SingleSelectListState
...
-
   return {
     tabListProps: {
-      ...mergeProps(focusWithinProps, collectionProps),
+      ...collectionProps,
       role: 'tablist',
-      'aria-disabled': isDisabled,
       'aria-label': ariaLabel,
-      tabIndex: isDisabled ? null : tabIndex
+      tabIndex: undefined
```
Comment (snowystinger) [inline packages/@react-aria/tabs/src/useTabs.ts:71]: can we just omit this property? or is something else setting it in collectionProps?

Comment (majornista) [reply packages/@react-aria/tabs/src/useTabs.ts:71]: I think collectionProps supplies a tabIndex.

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js (hunk 1 above)
Comment (majornista) [reply packages/@react-spectrum/tabs/stories/Tabs.stories.js:25]: A tablist doesn't require an `aria-label` attribute, but without an ancestor landmark element like `<main>`, which is missing in our Storybook example pages, I noticed that with VoiceOver the first tab to receive focus announces not only the focused tab, but all the other tabs as well. VoiceOver is announcing the nearest named container element or landmark region to provide context. If we add a `<main>`, VoiceOver announces the focused tab and the `<main>` landmark for context. However, with no `aria-label` or ancestor landmark, VoiceOver announces the contents of the tablist for context.

**Comment by majornista** [comment]
> I'm not sure why, but in 'Default' story, however I select "Tab 1" the onChange says empty string instead of 'val1'
> It appears to be a regression because I see the correct behavior on main.
> 
> I managed to get into a state where I could tab to the tabs at all. I could get to the content, but not the tabs. I can't make it happen again on purpose yet, I'll let you know if I do manage to reproduce it, but figured I'd bring it up anyways.

@snowystinger I haven't been able to reproduce either the onChange behavior you describe or, with the exception of the "disable all tabs" story, get into the state where I cannot focus the tabs at all.

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/tabs/stories/Tabs.stories.js:25]: Should we place a `main` tag in our stories then? or how do we expect people to use tabs? And what do we want to do in our documentation site to communicate this? I don't think we can have a bunch of `main`'s?

**Comment by snowystinger** [comment]
I don't know if it was the merge from master, but everything is behaving better now. Even VO for some reason. Maybe my CPU was bogged down with VO earlier.

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js (hunk 1 above)
Comment (majornista) [reply packages/@react-spectrum/tabs/stories/Tabs.stories.js:25]: We can the main tag to the storybook story template.

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**